<a href="https://colab.research.google.com/github/andresvillamayor/TDA/blob/master/simplicial_complexes_from_distance.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**simplicial_complexes_from_distance**

Andy Villamayor 

Complejos simpliciales desde una matriz de distancia 

In [1]:
#  Instalar gudhi
!pip install gudhi

     |████████████████████████████████| 28.2 MB 22 kB/s 


In [2]:
#  Importar librerias 
import pandas as pd
import numpy as np
from pylab import*
import gudhi as gd 
from  google.colab import drive

In [3]:
#  montar el drive para obtener el archivo
drive.mount("/content/drive")


Mounted at /content/drive


In [4]:
#  leer en archivo desde el drive
data = pd.read_csv("/content/drive/My Drive/TDA/distancia_total.csv", sep = "," , engine="python")

In [5]:
data.head()

0      1      2      3  ...      646      647      648      649
0  0.000  3.367  5.338  5.665  ...  741.325  742.839  744.794  746.039
1  3.367  0.000  4.389  4.674  ...  739.940  741.236  742.741  745.092
2  5.338  4.389  0.000  3.419  ...  740.289  740.649  741.904  743.561
3  5.665  4.674  3.419  0.000  ...  738.700  739.024  742.211  743.972
4  6.109  5.716  5.827  4.232  ...  737.778  739.328  741.143  742.414

[5 rows x 650 columns]

Filtración Vietoris-Rips de matriz de distancia 

La funcion RipsComplex crea el esqueleto
parametro max_edge_length es el diametro maximo

In [12]:
skeleton_distance = gd.RipsComplex(
    distance_matrix = data.values, 
    max_edge_length = 4
)

Se crea el complejo simplicial de RIPS a partir del esquelto , usando dimension = 2

In [13]:
Rips_simplex_tree_distance = skeleton_distance.create_simplex_tree(max_dimension=2)

El parametro max_dimension es la dimension maxima de los simplejos la funcion debe debolver el valor dado en la creacion del RIPS

In [14]:
Rips_simplex_tree_distance.dimension()

2

para hallar la dimension 2 se coloco el max_edge_length  = 4  

El numero de vertices 650  son los puntos de la matriz 

In [16]:
Rips_simplex_tree_distance.num_vertices()

650

Aca esta raro que no se haya aumentado !!!!!! que no haya aumentado el numero de simplejos
deberia ser mayor a 915

In [17]:
Rips_simplex_tree_distance.num_simplices()

915

In [18]:
rips_filtration = Rips_simplex_tree_distance.get_filtration()
rips_list = list(rips_filtration)
len(rips_list)

915

In [20]:
for splx in rips_list[0:800] :
    print(splx)

([0], 0.0)
([1], 0.0)
([2], 0.0)
([3], 0.0)
([4], 0.0)
([5], 0.0)
([6], 0.0)
([7], 0.0)
([8], 0.0)
([9], 0.0)
([10], 0.0)
([11], 0.0)
([12], 0.0)
([13], 0.0)
([14], 0.0)
([15], 0.0)
([16], 0.0)
([17], 0.0)
([18], 0.0)
([19], 0.0)
([20], 0.0)
([21], 0.0)
([22], 0.0)
([23], 0.0)
([24], 0.0)
([25], 0.0)
([26], 0.0)
([27], 0.0)
([28], 0.0)
([29], 0.0)
([30], 0.0)
([31], 0.0)
([32], 0.0)
([33], 0.0)
([34], 0.0)
([35], 0.0)
([36], 0.0)
([37], 0.0)
([38], 0.0)
([39], 0.0)
([40], 0.0)
([41], 0.0)
([42], 0.0)
([43], 0.0)
([44], 0.0)
([45], 0.0)
([46], 0.0)
([47], 0.0)
([48], 0.0)
([49], 0.0)
([50], 0.0)
([51], 0.0)
([52], 0.0)
([53], 0.0)
([54], 0.0)
([55], 0.0)
([56], 0.0)
([57], 0.0)
([58], 0.0)
([59], 0.0)
([60], 0.0)
([61], 0.0)
([62], 0.0)
([63], 0.0)
([64], 0.0)
([65], 0.0)
([66], 0.0)
([67], 0.0)
([68], 0.0)
([69], 0.0)
([70], 0.0)
([71], 0.0)
([72], 0.0)
([73], 0.0)
([74], 0.0)
([75], 0.0)
([76], 0.0)
([77], 0.0)
([78], 0.0)
([79], 0.0)
([80], 0.0)
([81], 0.0)
([82], 0.0)
([83], 0.0)
([